aea/cli/launch.py

```diff
@@ -81,10 +85,43 @@ def _launch_subprocesses(click_context: click.Context, agents: List[Path]):
     sys.exit(failed)
 
 
+def _launch_threads(click_context: click.Context, agents: List[Path]):
+    """
+    Launch many agents, multithreaded.
+
+    :param agents: the click context.
+    :param agents: list of paths to agent projects.
+    :return: None
+    """
+    aeas = []  # type: List[AEA]
+    for agent_directory in agents:
+        aeas.append(AEABuilder.from_aea_project(agent_directory).build())
+
+    threads = [Thread(target=agent.start) for agent in aeas]
+    for t in threads:
+        t.start()
+
+    try:
+        for t in threads:
+            t.join()
+    except KeyboardInterrupt:
+        logger.info("Keyboard interrupt detected.")
+    finally:
+        for idx, agent in enumerate(aeas):
+            if not agent.liveness.is_stopped:
+                agent.stop()
+                threads[idx].join()
```

Comment:
Would it be safer to move the join method outside the if clause? It could be that a thread is lingering even if the the liveness is stopped?

Reply:
yes, well spotted.